MovingBlocks/Terasology
# build(gradle): upgrade gradle 6.4.2 >>> 8.2.1

### Contains

* Updating gradle from 6.4.2 -> 8.2.1
* Updating gradle's files for which became invalid

### How to test

* run `gradlew jar game` -> should build and run game
* run `gradlew unitTest` -> should run the unit test suites

### Remarks

Java 20 is not yet supported, see https://github.com/gradle/gradle/issues/23488.

Closes #5111
